### PR TITLE
Feature: Use the current working directory (of the opened buffer) to execute the git commands

### PR DESCRIPTION
**Pull request by mdedonno1337**
Sorry for the new PR, I've messed up my repo ...

This change is the same as the last one in the PR "Set the working directory for the git commands - add support for submodules " #49

### CODE REVIEW

**Review by stsewd — COMMENTED** [review]
From your comment, then using `-C` seems right, but let's rename this to `fzf_checkout_use_current_buf_cwd`  or `fzf_checkout_use_buf_cwd` as this isn't only helpful for submodules :)

File: autoload/fzf_checkout.vim [hunk 1]
```diff
@@ -280,3 +298,12 @@ function! fzf_checkout#complete_branches(arglead, cmdline, cursorpos) abort
 
   return join(l:options, "\n")
 endfunction
+
+function! fzf_checkout#get_wd() abort
+  if get(g:, 'fzf_checkout_enable_submodules', v:false)
+    return '-C '.shellescape(expand('%:p:h'))
```
[the remark this answers is not in the document]
Comment (mdedonno1337) [reply autoload/fzf_checkout.vim:304]: I had in mind very old versions of git, before 2013 (https://github.com/git/git/blob/be7935ed8bff19f481b033d0d242c5d5f239ed50/git.c#L290 , commit https://github.com/git/git/commit/44e1e4d67d5148c245db362cc48c3cc6c2ec82ca).

Probably useless keep support for people stuck in 2013.

**Comment by mdedonno1337** [comment]
Should be OK this time. Let me know otherwise.

File: autoload/fzf_checkout.vim [hunk 2]
```diff
@@ -94,6 +94,7 @@ function! fzf_checkout#execute(type, action, lines) abort
   let l:Execute_command = l:actions[l:action]['execute']
   if type(l:Execute_command) == v:t_string
     let l:Execute_command = substitute(l:Execute_command, '{git}', g:fzf_checkout_git_bin, 'g')
+    let l:Execute_command = substitute(l:Execute_command, '{wd}', fzf_checkout#get_cwd(), 'g')
```
Comment (stsewd) [inline autoload/fzf_checkout.vim:97]: Let's use cwd here as well

```suggestion
    let l:Execute_command = substitute(l:Execute_command, '{cwd}', fzf_checkout#get_cwd(), 'g')
```

File: doc/fzf-checkout.txt [hunk 3]
```diff
@@ -103,6 +103,9 @@ and the value is a dictionary with the following keys:
 - `execute`: A string to be executed when performing this action.
   You can make use of the following placeholders:
   - `{git}`: Replaced with the value from |g:fzf_checkout_git_bin|
+  - `{wd}`: Replaced, if |g:fzf_checkout_use_current_buf_cwd| is enabled, with
+    the correct working directory. This option is manadged automatically by
+    the plugin.
```
Comment (stsewd) [inline doc/fzf-checkout.txt:108]:
```suggestion
    the directory from the current buffer or the current working directory.
```

File: doc/fzf-checkout.txt [hunk 4]
```diff
@@ -209,5 +212,16 @@ Set it to `v:false` if you want to have full control over the defined actions.
...
    let g:fzf_checkout_merge_settings= v:true
 <
+                                          *g:fzf_checkout_use_current_buf_cwd*
+g:fzf_checkout_use_current_buf_cwd~
 
+Set the |g:fzf_checkout_use_current_buf_cwd| variable to operate in the
+correct directory for the current buffer. This will execute the `git` commands
+in the folder containing the file displayed in the current buffer. This
+facilitate the work in a multi-git-repository project (with submodules,...).
+This option is disable by default.
```
Comment (stsewd) [inline doc/fzf-checkout.txt:222]:
```suggestion
If true, set the `{cwd}` placeholder to the directory from the current buffer.
Useful when working in a multi-git-repository project (with submodules for example).
This option is disable by default.
```

File: doc/fzf-checkout.txt [hunk 5]
```diff
@@ -209,5 +212,16 @@ Set it to `v:false` if you want to have full control over the defined actions.
...
+
+>
+    let g:fzf_checkout_use_current_buf_cwd = v:true
```
Comment (stsewd) [inline doc/fzf-checkout.txt:225]: I'm using the codeblocks to show the defaults

```suggestion
    let g:fzf_checkout_use_current_buf_cwd = v:false
```

**Review by stsewd — COMMENTED** [review]
Left some suggestions to make the docs shorter

**Comment by stsewd** [comment]
> Should be OK this time. Let me know otherwise.
> Is it possible to change the merge message on your side, or have I to rename the branch so the merge message does not contain "submodule" but something more generic?

Yeah, sure, I can just squash and change the commit message when this is ready, so don't worry about that.

File: autoload/fzf_checkout.vim [hunk 6]
```diff
@@ -280,3 +298,12 @@ function! fzf_checkout#complete_branches(arglead, cmdline, cursorpos) abort
 
   return join(l:options, "\n")
 endfunction
+
+function! fzf_checkout#get_cwd() abort
+  if get(g:, 'fzf_checkout_use_current_buf_cwd', v:false)
```
Comment (stsewd) [inline autoload/fzf_checkout.vim:303]: defaults are defined in plugin/fzf_checkout.vim

**Review by stsewd — APPROVED** [review]
Thanks for implementing this!

**Pull request merged**